macos/Classes/FilePickerPlugin.swift

```diff
@@ -0,0 +1,123 @@
+import Cocoa
+import FlutterMacOS
+
+public class FilePickerPlugin: NSObject, FlutterPlugin {
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(
+      name: "miguelruivo.flutter.plugins.filepicker",
+      binaryMessenger: registrar.messenger)
+    let instance = FilePickerPlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult)
+  {
+    switch call.method {
+    case "pickFiles":
+      handleFileSelection(call, result: result)
+
+    case "getDirectoryPath":
+      handleDirectorySelection(call, result: result)
+
+    case "saveFile":
+      handleSaveFile(call, result: result)
+
+    default:
+      result(FlutterMethodNotImplemented)
+    }
+  }
+
+  private func handleFileSelection(
+    _ call: FlutterMethodCall, result: @escaping FlutterResult
+  ) {
+    let dialog = NSOpenPanel()
+    let args = call.arguments as! [String: Any]
+
+    dialog.showsResizeIndicator = true
+    dialog.directoryURL = URL(
+      fileURLWithPath: args["initialDirectory"] as? String ?? ""
+    )
+    dialog.showsHiddenFiles = false
+    dialog.allowsMultipleSelection = args["allowMultiple"] as? Bool ?? false
+    dialog.canChooseDirectories = false
+    dialog.allowedFileTypes = args["allowedExtensions"] as? [String] ?? []
```

Comment:
This property is deprecated: https://developer.apple.com/documentation/appkit/nssavepanel/allowedfiletypes

Can you switch to the replacement on newer MacOS versions, using an availability check?

Reply:
updated and unified for pick and save